src/app/qgsattributesforminitcode.cpp

```diff
@@ -0,0 +1,84 @@
+#include "qgsattributesforminitcode.h"
+#include "ui_qgsattributesforminitcode.h"
+#include "qgssettings.h"
+
+#include <QFileDialog>
+
+QgsAttributesFormInitCode::QgsAttributesFormInitCode()
+{
+  setupUi( this );
+
+  // Init function stuff
+  mInitCodeSourceComboBox->addItem( tr( "" ) );
+  mInitCodeSourceComboBox->addItem( tr( "Load from external file" ) );
+  mInitCodeSourceComboBox->addItem( tr( "Provide code in this dialog" ) );
+  mInitCodeSourceComboBox->addItem( tr( "Load from the environment" ) );
+
+  connect( mInitCodeSourceComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsAttributesFormInitCode::mInitCodeSourceComboBox_currentIndexChanged );
+  connect( pbtnSelectInitFilePath, &QToolButton::clicked, this, &QgsAttributesFormInitCode::pbtnSelectInitFilePath_clicked );
+
+}
+
+QgsAttributesFormInitCode::~QgsAttributesFormInitCode()
+{
+}
+void QgsAttributesFormInitCode::setCodeSource( QgsEditFormConfig::PythonInitCodeSource initCodeSource )
+{
+  mInitCodeSourceComboBox->setCurrentIndex( initCodeSource );
+  mInitCodeSourceComboBox_currentIndexChanged( mInitCodeSourceComboBox->currentIndex() );
+
+}
+void QgsAttributesFormInitCode::setInitFunction( const QString &initFunction )
+{
+  mInitFunctionLineEdit->setText( initFunction );
+}
+void QgsAttributesFormInitCode::setInitFilePath( const QString &initFilePath )
+{
+  mInitFilePathLineEdit->setText( initFilePath );
+}
+void QgsAttributesFormInitCode::setInitCode( const QString &initCode )
+{
+  mInitCodeEditorPython->setText( initCode );
+}
+
+QgsEditFormConfig::PythonInitCodeSource QgsAttributesFormInitCode::codeSource() const
+{
+  return ( QgsEditFormConfig::PythonInitCodeSource )mInitCodeSourceComboBox->currentIndex();
```

Comment:
Maybe use currentData here

    return mInitCodeSourceComboBox->currentData()

and in the initialization

    addItem->( text, sourceType )